### ISSUE TITLE
[Bug] Error after upgrading to pyclesperanto without prototype

### ISSUE THREAD

**Issue by haesleinhuepf** [clEsperanto/pyclesperanto#255]
**Pre-requist tasks**
- [x] My device drivers are up to date.
- [x] The [troublemaking solutions](https://github.com/clEsperanto/pyclesperanto?tab=readme-ov-file#troubleshooting-graphics-cards-drivers) did not solved my issue.

**Describe the bug**
A clear and concise description of what the bug is. What were you trying to do and what happened.

I upgraded a napari-plugin to use pyclesperanto instead of the _prototype version. There [tests](https://github.com/clEsperanto/napari_pyclesperanto_assistant/actions/runs/11275221809/job/31356142771?pr=73) are failing now, presumably because of errors in pyclesperanto.

**To Reproduce**
Steps to reproduce the behavior, ideally a notebook ready to run, with test data, at minima a code snippet.

Download the "[pyclesperanto-without-prototype](https://github.com/clEsperanto/napari_pyclesperanto_assistant/tree/pyclesperanto-without-prototype)" branch and run "pytest" from the root dir.

## Error 1: A parameter was renamed from "maximum_size" to "max_size":
```
>       exclude_small_labels(labels)
...
E               TypeError: got an unexpected keyword argument 'maximum_size'
```

I propose to **undo the renaming on CLIc side.**

## Error 2: Object count is different
```
    def test_labeling_and_statistics():
        from skimage.io import imread
        image = imread("napari_pyclesperanto_assistant/data/blobs.tif")

        from napari_pyclesperanto_assistant._napari_cle_functions import  voronoi_otsu_labeling
        labels = voronoi_otsu_labeling(image, spot_sigma=3.5)

        from napari_pyclesperanto_assistant._statistics_of_labeled_pixels import statistics_of_labeled_pixels
        stats = statistics_of_labeled_pixels(image, labels)

>       assert len(stats) == 65
E       assert 58 == 65
```
This might be harder to fix, but I also recall that blobs.tif has 65 objects, not 58.

 

**Expected behavior**
A clear and concise description of what you expected to happen. Mainly if the issues is not a crash but a bad behaviour of a filter or operation.

Tests pass.

**Screenshots**
If applicable, add screenshots to help explain your problem. Expected results, vs actuall results. Or the output error message if any.

**Configuration details:**
 - OS:  Windows 11
 - Device:  NVIDIA RTX 4070
 - Device type:dedicated GPU

**Additional context**

**Comment by StRigaud** [comment]
Hi @haesleinhuepf,

Is this the only one which make the assistant test fail? High chances that not all the parameter names are exactly the same than the prototype. 

Not sure I will have the courage to chase and fix all, so will only focus on major one and demo/script breakers if that's okay?

**Comment by haesleinhuepf** [comment]
> Not sure I will have the courage to chase and fix all

As discussed in other places: I think it is extremely important to have the API as backwards compatible as possible. Otherwise, people may not switch to the non-prototype.

**Comment by StRigaud** [comment]
> I think it is extremely important to have the API as backwards compatible as possible

I agree, simply saying that I might need some help to see and find all these little differences, and if it conserne small code change, a direct PR might be quicker :)

Anyway, I've made a PR to fix this in CLIc. It will impact pyclesperanto in the next release (do I wait wednesday to include other possible code update or do I release asap?)

**Comment by haesleinhuepf** [comment]
No hurry from my side. We can try together on Wednesday.

Did you also solve Error 2?

**Comment by StRigaud** [comment]
> Did you also solve Error 2?

oups, forgot about this one! 

Do we have a lead on where the issue could come from? I will have a look when I get a bit more than 5 min otherwise monday.

**Comment by haesleinhuepf** [comment]
No idea. I would first look into image data types along the intermediate results of the workflow

**Comment by StRigaud** [comment]
Ok, the issue seems related to the `detect_maxima`. I am loosing spots on the edges.

This is kind of related to this issue which was never properlly addressed / fixed / clarrified:
- https://github.com/clEsperanto/clij-opencl-kernels/issues/23

The initial issue was the top left corner and detection in 3D. We are not using the same version, I am not sure your version is bugless but mine is clearly not good.  I am loosing spots along some of the right and bottom edge of the image. 

If you can have a look at the kernel and issue, that would be great, otherwise I will focus on this issue when I'm back on Monday.

**Comment by StRigaud** [comment]
Both issues should be now fix with the incoming release of [0.13.4 version](https://github.com/clEsperanto/pyclesperanto/releases)
